### ISSUE TITLE
MPI error - CASPT2 geometry optimization

### ISSUE THREAD

**Issue by aerask** [nubakery/bagel#126]
Dear developers,

I'm running CASPT2 geometry optimizations on a variety of transition metal complexes and am consistently getting an error before any geometries converge. Although consistent in stopping computations, this error is inconsistent in timing with respect to complex and active space. I've attached some files for doing a CAS(2,2)PT2 geometry optimization on a di-copper complex as an example. 

[cuaqac.zip](https://github.com/nubakery/bagel/files/2325984/cuaqac.zip)
Submission script: bagel.sbatch
Input: cuaqac.2_2.12.1.12.json
Output: cuaqac.2_2.12.1.12.saun (.saun is not important, it's just an inside joke)
Error: jcuaqac.2_2.12.1.12.2738529
User defined basis sets used: 6-31gs.turbo.json and rimp2-vdz.gaus.json

The first, second, and third numbers after 2_2 in file names denote tasks/node, mpi threads/process, and processes/task respectively (set in bagel.sbatch). I've also tried submitting jobs with fewer processes per task (eg. 12.2.6) but run into the same error. It tends to fail by around optimization iteration 30 using this system, basis, and active space (around iter 6 for CAS(4,4)). Do you know what might be causing this and/or do you have any suggestions to fix it?

**Comment by shiozaki** [comment]
Hi Jae Woo,

Is there any chance you can take a look at this?

Toru

**Comment by jwpk1201** [comment]
Hi Alan,

Can you send me the optimization log file (opt.log)? I will take a look at it.

Sincerely,
Jae Woo

**Comment by aerask** [comment]
After a bit of experimentation behind the scenes, we have a solution. Intel OpenMPI sometimes leaks memory and therefore doesn't handle CASSCF iterations very well. The solution to this is quite simple and just requires a change of strategies. Before I was using an equivalent number of tasks/node and processes/task, with just 1 mpi thread/process. Instead of that, using an equivalent number of tasks/node and mpi threads/process, with just 1 process/task solves the issue. 

Simply stated, the change in my submission script simply changes the similar lines to the ones below:
#SBATCH --ntasks-per-node=12
export BAGEL_NUM_THREADS=12
export MKL_NUM_THREADS=12
mpirun -n 1 --bind-to none BAGEL file.json > file.saun

I've been able to run a variety of tests with this up to ~200 electrons using CAS(2,2) and (4,4). 

Thanks for the help!
Alan

**Comment by Piju123** [comment]
Dear developers,

I want to running a NACME calculation using CASPT2(6,5) method. For this I have used 
nodes=1
ntasks-per-node=16
cpus-per-task=1
mem-per-cpu=16GB
 export BAGEL_NUM_THREADS=12
export MKL_NUM_THREADS=12

But the main problem is at the position of CASPT2 iteration, the jobs get killed. I think this should be due to the memory issue. So, I have also tried to run same things but this time increasing the nstasks-per-node as well as the nodes to supply more memory. But again, the job gets killed.  Therefore, what will be the way out.

with regrads,
Pijush
